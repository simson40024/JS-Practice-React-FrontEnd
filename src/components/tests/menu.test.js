import Menu from "../header/menu";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

describe("Menu", () => {
  it("shoud create Menu", () => {
    act(() => {
      createRoot(container).render(
        <BrowserRouter>
          <Menu />
        </BrowserRouter>
      );
    });
    expect(container).toMatchSnapshot();
  });
});
