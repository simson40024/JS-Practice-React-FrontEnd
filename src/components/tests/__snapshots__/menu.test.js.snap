// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu shoud create Menu 1`] = `
<div>
  <nav
    class="main-menu"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Shop
        </a>
      </li>
      <li>
        <a
          href="/cart"
        >
          Cart
        </a>
      </li>
      <li>
        <a
          href="/register"
        >
          Register
        </a>
      </li>
      <li>
        <a
          href="/sign"
        >
          SignIn
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
